.github/workflows/create-release.yml

```
@@ -27,10 +27,23 @@ jobs:
           python-version: 3.8
       - name: Change version number
         id: version
+        # See this issue for explanation and testing:
+        # https://github.com/cmu-delphi/delphi-epidata/pull/1473
         run: |
           python -m pip install bump2version
-          echo -n "::set-output name=next_tag::"
-          bump2version --list ${{ github.event.inputs.versionName }} | grep new_version | sed -r s,"^.*=",,
+          allowed_pattern="^(major|minor|patch|[0-9]+\.[0-9]+\.[0-9]+)$"
+
+          if [[ ! ${{ github.event.inputs.versionName }} =~ $allowed_pattern ]]; then
+            echo "\nInvalid version name: ${{ github.event.inputs.versionName }}"
+            exit 1
+          fi
+
+          if [[ ${{ github.event.inputs.versionName }} =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            NEXT_TAG="${{ github.event.inputs.versionName }}"
+          else
+            NEXT_TAG=$(bump2version --dry-run --list ${{ github.event.inputs.versionName }} | grep ^new_version | sed -r s,"^.*=",,)
+          fi
+          echo "next_tag=$NEXT_TAG" >> $GITHUB_OUTPUT
```

we need to make sure the new version gets applied:
```suggestion
          if [[ ${{ github.event.inputs.versionName }} =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
            # use given version number
            NEXT_TAG="${{ github.event.inputs.versionName }}"
          else
            # calculate new version number based on given tag
            NEXT_TAG=$(bump2version --dry-run --list ${{ github.event.inputs.versionName }} | grep ^new_version | sed -r s,"^.*=",,)
          fi
          # apply given or calculated version number
          bump2version --new-version $NEXT_TAG _ignored_arg_
          # save version number for later
          echo "next_tag=$NEXT_TAG" >> $GITHUB_OUTPUT
```

---

We never actually applied the version in this step before, so I don't think is right. The actual version bump happens in the  "Bump Python versions if client files changed" step below.

---

we did, it was in old line 33: `bump2version --list ${{ github.event.inputs.versionName }} | grep new_version | sed -r s,"^.*=",,`

the step below is just for the client version, this is for the server and acquisition code.

---

That line doesn't actually bump version, it just extracts the version number out and I'm already doing it in new line 44. The 
```
          # apply given or calculated version number
          bump2version --new-version $NEXT_TAG _ignored_arg_
```
in your suggestion was never done in the original action in this step and I don't think we need it. Are you intending to change the behavior of the original CI here? This PR is just about fixing the CI argument validation.

---

Ah I misunderstood, thought that `bump2version --list ${{ github.event.inputs.versionName }} | grep new_version | sed -r s,"^.*=",,` didn't actually operate on the files and make changes.